adsputils/__init__.py

```diff
@@ -424,12 +409,36 @@ def _set_serializer(self):
         self.conf['CELERY_TASK_SERIALIZER'] = 'adsmsg'
         self.conf['CELERY_RESULT_SERIALIZER'] = 'adsmsg'
 
-    def forward_message(self, *args, **kwargs):
-        """Class method that is replaced during initializiton with the real
-        implementation (IFF) the OUTPUT_TASKNAME and oother OUTPUT_ parameters
-        are specified."""
-        if not self.forwarding_connection or not self._forward_message:
-            raise NotImplementedError('Sorry, your app is not properly configured.')
+    def forward_message(self, output_taskname=None, output_celery_broker=None, *args, **kwargs):
+        """Class method that sets up the message forwarding handler dynamically based on
+        OUTPUT_TASKNAME and OUTPUT_CELERY_BROKER."""
+
+        # Use OUTPUT_CELERY_BROKER from config if not provided at call time
+        broker = output_celery_broker or self._config.get('OUTPUT_CELERY_BROKER')
+        if broker:
+            # kombu connection is lazy loaded, so it's ok to create now
+            self.forwarding_connection = BrokerConnection(broker)
```

Comment:
I was initially concerned that this would create a lot of extra connections but because of the lazy nature of the connection I think we should be okay to do this all in `forward_message`.